include/nbl/builtin/hlsl/math.hlsl

```diff
@@ -0,0 +1,30 @@
+// Copyright (C) 2022 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#ifndef _NBL_BUILTIN_HLSL_MATH_INCLUDED_
+#define _NBL_BUILTIN_HLSL_MATH_INCLUDED_
+
+#include "nbl/builtin/hlsl/cpp_compat/type_traits.h"
+
+// REVIEW-519: Review this whole header and content (whether it should be here or somewhere else)
+namespace nbl
+{
+namespace hlsl
+{
+namespace math
```

Comment:
there's already a `math` header in C++ STL which is for normal (albeit with constexpr and consteval) math functions, call it `meta_math`